src/machine/system_calls.rs

```diff
@@ -4216,23 +4215,21 @@ impl Machine {
 
     #[inline(always)]
     pub(crate) fn maybe(&mut self) {
-        fn generate_random_bits(num_bits: usize) -> u64 {
```

Comment:
Not commenting on the PR in particular, but what the heck is going on here?

We're defining a function that can generate any number of bits, then only ever invoking it to generate one bit. (?!) Could this not be a single line?

```rust
self.machine_st.fail = (&mut self.rng).gen_bool(0.5);

// Also, why are floats involved? Use gen_range instead?
self.machine_st.fail = (&mut self.rng).gen_range(0,2) == 0;
```